script/signup.js

```diff
@@ -0,0 +1,105 @@
+const inputEmail = document.querySelector("#inputEmail");
+const inputPw = document.querySelector("#inputPw");
+const inputPwConfirm = document.querySelector("#inputPwConfirm");
+const emailError = document.querySelector("#email_err");
+const pwError = document.querySelector("#pw_err");
+const pwConfirmError = document.querySelector("#pwConfirm_err");
+const signupBtn = document.querySelector("#signupBtn");
+const inputPwIcon = document.querySelector("#inputPwIcon");
+const inputPwComfirmIcon = document.querySelector("#inputPwComfirmIcon");
+
+let emailValid = false;
+let passwordValid = false;
+let passwordConfirmValid = false;
+
+function checkEmail() {
+    const email_format =
+        /^([0-9a-zA-Z_\.-]+)@([0-9a-zA-Z_-]+)(\.[0-9a-zA-Z_-]+){1,2}$/;
+    if (!inputEmail.value) {
+        emailError.textContent = "이메일을 입력해주세요.";
+        inputEmail.className = "error-box";
+    } else if (!email_format.test(inputEmail.value)) {
+        emailError.textContent = "올바른 이메일 주소가 아닙니다.";
+        inputEmail.className = "error-box";
+    } else if (inputEmail.value === "test@codeit.com") {
+        emailError.textContent = "이미 사용 중인 이메일입니다.";
+    } else {
+        emailError.textContent = " ";
+        inputEmail.className = "inputbox";
+        emailValid = true;
+    }
+}
+
+function checkPassword() {
+    const passwordRegex = /^(?=.*[a-zA-Z])(?=.*[0-9]).{8,16}$/;
+    if (!inputPw.value) {
+        pwError.textContent = "비밀번호를 입력해주세요.";
+        inputPw.className = "error-box";
+    } else if (!passwordRegex.test(inputPw.value)) {
+        pwError.textContent =
+            "비밀번호는 영문, 숫자 조합 8자 이상 입력해 주세요.";
+        inputPw.className = "error-box";
+    } else if (inputPw.value && passwordRegex.test(inputPw.value)) {
+        pwError.textContent = " ";
+        inputPw.className = "inputbox";
+        passwordValid = true;
+    }
+}
+
+function passwordConfirm() {
+    if (!inputPwConfirm.value) {
+        pwConfirmError.textContent = "비밀번호를 입력해주세요.";
+        inputPwConfirm.className = "error-box";
+    } else if (inputPwConfirm.value !== inputPw.value) {
+        pwConfirmError.textContent = "비밀번호가 일치하지 않아요.";
+    } else {
+        pwConfirmError.textContent = " ";
+        inputPwConfirm.className = "inputbox";
+        passwordConfirmValid = true;
+    }
+}
+
+function signUp() {
+    if (
+        emailValid === true &&
+        passwordValid === true &&
+        passwordConfirmValid === true
+    ) {
+        location.href = "/folder";
```

Comment:
location.href는 데이터 전송 없이 브라우저의 주소창을 변경하거나 새로운 페이지로 이동 할 때 사용하는 것으로 알고있습니다. form 태그에 id를 부여한 후 데이터를 전송하는 방식을 사용해보는 건 어떨까요?
```suggestion
const form = document.getElementById('#form');
form.action = '/folder.html';
form.method = 'POST';
form.submit();
```

Reply:
c46f41b7543040621436ca4e1d6cd5e2dda536c3 생각하지 못한 부분인데 알려주셔서 감사합니당! 그렇게 수정했습니다!!